File: abc/010/3.py
import math
def dist(p1, p2):
    dx = abs(p1[0]-p2[0])
    dy = abs(p1[1]-p2[1])
    return math.sqrt(dx*dx + dy*dy)

x0, y0, x1, y1, T, V = map(int, raw_input().split())
n = int(raw_input())
P = [] * n
for i in range(n):
    x, y = map(int, raw_input().split())
    d = dist((x, y), (x0, y0)) + dist((x, y), (x1, y1))
    if (d <= T * V):
        print 'YES'
        exit()
print 'NO'

File: abc/010/3.hs
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE ScopedTypeVariables #-}
import Control.Monad
import Control.Applicative
import Debug.Trace
import qualified Data.ByteString.Char8 as BC
import Data.Maybe (fromJust)

readInts = map (fst . fromJust . BC.readInt) . BC.words <$> BC.getLine

dist (x1, y1) (x2, y2) = sqrt $ x' * x' + y' * y'
    where
        x' = x1 - x2
        y' = y1 - y2

canMove xa ya xb yb t v ([x, y] :: [Double])
    = dist (xa, ya) (x, y) + dist (xb, yb) (x, y) <= t * v

main = do
    [xa, ya, xb, yb, t, v] <- map fromIntegral <$> readInts
    [n] <- readInts
    ans <- replicateM n $ canMove xa ya xb yb t v <$> map fromIntegral <$> readInts
    putStrLn $ if or ans then "YES" else "NO"
